### PR TITLE
[AIRFLOW-1995] add on_kill method to SqoopOperator

### PR DESCRIPTION
**Pull request by Acehaidrey**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow 1995](https://issues.apache.org/jira/browse/AIRFLOW-1995/) issues and references them in the PR title. For example, "[AIRFLOW-1995] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1995

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
The sqoop operator behind the scenes sets off a bash command, but one bug arises due to the fact that the on kill method is not implemented. It needs to be implemented in order to kill the process for a sqoop task, otherwise weird things happen trying to end the task.

CC @Fokko 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Difficult to get a test for. Tested locally by making sure the pid is killed as soon as method is called locally but without binary it fails. I use same on kill method as bash operator

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/hooks/sqoop_hook.py [hunk 1]
```diff
@@ -93,6 +93,7 @@ def Popen(self, cmd, **kwargs):
                               stdout=subprocess.PIPE,
                               stderr=subprocess.STDOUT,
                               **kwargs)
+        self.sp = sp
```
Comment (Fokko) [inline airflow/contrib/hooks/sqoop_hook.py]: Can we do `self.sp = subprocess.` directly and reference everything to `self.sp`? This makes it a bit simpler.

**Review by Fokko — CHANGES_REQUESTED** [review]
I have one comment and please add a test that invokes the on_kill method:
https://github.com/apache/incubator-airflow/blob/master/tests/contrib/hooks/test_sqoop_hook.py

File: airflow/contrib/hooks/sqoop_hook.py (hunk 1 above)
Comment (Acehaidrey) [reply airflow/contrib/hooks/sqoop_hook.py]: Hey Fokko, thanks for the response. Will go ahead and do that. Agreed it keeps it simpler

**Comment by Fokko** [comment]
Thanks @Acehaidrey !

**Pull request closed without merging**